src/frontend/src/modals/IOModal/components/chatView/newChatView.tsx

```diff
@@ -135,53 +142,54 @@ export default function ChatView({
       onDrop={onDrop}
     >
       <div ref={messagesRef} className="chat-message-div">
-        {lockChat || chatHistory?.length > 0 ? (
-          chatHistory.map((chat, index) => (
-            <ChatMessage
-              setLockChat={setLockChat}
-              lockChat={lockChat}
-              chat={chat}
-              lastMessage={chatHistory.length - 1 === index ? true : false}
-              key={`${chat.id}-${index}`}
-              updateChat={updateChat}
-              closeChat={closeChat}
-            />
-          ))
-        ) : (
-          <div className="flex h-full w-full flex-col items-center justify-center">
-            <div className="flex flex-col items-center justify-center gap-4 p-8">
-              {ENABLE_NEW_LOGO ? (
-                <LangflowLogo
-                  title="Langflow logo"
-                  className="h-10 w-10 scale-[1.5]"
-                />
-              ) : (
-                <ChainLogo
-                  title="Langflow logo"
-                  className="h-10 w-10 scale-[1.5]"
-                />
-              )}
-              <div className="flex flex-col items-center justify-center">
-                <h3 className="mt-2 pb-2 text-2xl font-semibold text-primary">
-                  New chat
-                </h3>
-                <p className="text-lg text-muted-foreground">
-                  <TextEffectPerChar>
-                    Test your flow with a chat prompt
-                  </TextEffectPerChar>
-                </p>
+        {chatHistory &&
```

Comment:
that was necessary because the chat history was getting assigned its value milliseconds after the component rendered. That caused the "New Chat" page to appear for a millisecond, like a glitch.

Reply:
That is when the chat already has some messages. So, before showing the messages, it was showing the "New Chat" screen for like 200ms